b2sdk/raw_api.py

```diff
@@ -407,9 +416,19 @@ def get_download_authorization(
             validDurationInSeconds=valid_duration_in_seconds
         )
 
-    def get_file_info(self, api_url, account_auth_token, file_id):
+    def get_file_info_by_id(self, api_url: str, account_auth_token: str,
+                            file_id: str) -> Dict[str, Any]:
         return self._post_json(api_url, 'b2_get_file_info', account_auth_token, fileId=file_id)
 
+    def get_file_info_by_name(
+        self, download_url: str, account_auth_token: str, bucket_name: str, file_name: str
+    ) -> Dict[str, Any]:
+        download_url = self.get_download_url_by_name(download_url, bucket_name, file_name)
+        response = self.b2_http.head_url(
```

Comment:
```Traceback (most recent call last):
  File "/home/maciek/Workspace/b2/b2-sdk-python/b2sdk/raw_api.py", line 775, in test_raw_api
    test_raw_api_helper(raw_api)
  File "/home/maciek/Workspace/b2/b2-sdk-python/b2sdk/raw_api.py", line 945, in test_raw_api_helper
    download_url, account_auth_token, bucket_name, file_name
  File "/home/maciek/Workspace/b2/b2-sdk-python/b2sdk/raw_api.py", line 427, in get_file_info_by_name
    response = self.b2_http.head_url(
AttributeError: 'B2Http' object has no attribute 'head_url'
```
Please run the integration test manually, as CI doesn't run it from forks

Reply:
What's required to run the integration test? I don't want to pay money to run a test.

Reply:
Good question. We have recently changed the CI and enabled integration tests, but for security reasons, the integration tests don't work from forks as they use GitHub secrets. We may use [pull_request_target](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target) in the future to enable the integration tests from forks, but until then contributors can use free B2 accounts which are more than enough to run those tests. Most of the tests in B2 SDK use the simulator anyway

Reply:
On running the tests locally, I found out also the SDK exception handling doesn't handle 404 very well (as discovered when the integration tests failed due to a dependent test that hid the file.... those tests should definitely be split out and some error cases tested...).
So I added two exception types to ensure that we don't mis-interpret those cases.